pkg/client/clientset/unversioned/extensions.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was autogenerated by the command:
+// $ /tmp/go-build881751838/command-line-arguments/_obj/exe/main
+// Do not edit it manually!
+
+package unversioned
+
+import (
+	"fmt"
+	unversioned "k8s.io/kubernetes/pkg/client/unversioned"
+	"strings"
+)
+
+type ExtensionsInterface interface {
+	DeploymentNamespacer
+}
+
+// ExtensionsClient is used to interact with features provided by the Extensions group.
+type ExtensionsClient struct {
+	*unversioned.RESTClient
+}
+
+func (c *ExtensionsClient) Deployments(namespace string) DeploymentInterface {
+	return newDeployments(c, namespace)
+}
+
+// NewExtensions creates a new ExtensionsClient for the given config.
+func NewExtensions(c *unversioned.Config) (*ExtensionsClient, error) {
```

Comment:
I think common client struct/function like `Config` and `DefaultKubernetesUserAgent()` should be kept at a common place, not repeatedly generated for each clientset.